core/src/main/kotlin/fr/sncf/osrd/api/api_v2/stdcm/STDCMEndpointV2.kt

```diff
@@ -200,6 +212,66 @@ class STDCMEndpointV2(private val infraManager: InfraManager) : Take {
     }
 }
 
+fun buildTemporarySpeedLimitManager(
+    infra: FullInfra,
+    speedLimits: Collection<STDCMTemporarySpeedLimit>
+): TemporarySpeedLimitManager {
+    var outputSpeedLimits: MutableMap<DirTrackChunkId, MutableList<TemporarySpeedLimit>> =
+        mutableMapOf()
+    for (speedLimit in speedLimits) {
+        var chunkIds = MutableDirStaticIdxArrayList<TrackChunk>()
+        var startOffsets = MutableOffsetArrayList<TrackChunk>()
+        var endOffsets = MutableOffsetArrayList<TrackChunk>()
+        var directions = mutableListOf<Direction>()
+        for (trackRange in speedLimit.trackRanges) {
+            val trackSection = infra.rawInfra.getTrackSectionFromName(trackRange.trackSection)
+            trackSection?.let {
+                val trackChunks = infra.rawInfra.getTrackSectionChunks(trackSection)
+                for (trackChunkId in trackChunks) {
+                    val trackChunkLength = infra.rawInfra.getTrackChunkLength(trackChunkId).distance
+                    val chunkStartOffset = infra.rawInfra.getTrackChunkOffset(trackChunkId)
+                    val chunkEndOffset = chunkStartOffset + trackChunkLength
+                    if (chunkEndOffset < trackRange.begin || trackRange.end < chunkStartOffset) {
+                        continue
+                    }
+                    var startOffset =
+                        Offset<TrackChunk>(
+                            Distance.max(0.meters, trackRange.begin - chunkStartOffset)
+                        )
+                    var endOffset =
+                        Offset<TrackChunk>(
+                            Distance.min(trackChunkLength, trackRange.end - chunkStartOffset)
+                        )
+                    var direction =
+                        when (trackRange.direction) {
+                            EdgeDirection.START_TO_STOP -> Direction.INCREASING
+                            EdgeDirection.STOP_TO_START -> Direction.DECREASING
+                        }
+                    val dirTrackChunkId = DirTrackChunkId(trackChunkId, direction)
+
+                    chunkIds.add(dirTrackChunkId)
+                    startOffsets.add(startOffset)
+                    endOffsets.add(endOffset)
+                    directions.add(direction)
+                }
+            }
+        }
+        val speedLimit: TemporarySpeedLimit =
+            TemporarySpeedLimit(
+                Speed.fromMetersPerSecond(speedLimit.speedLimit),
+                chunkIds,
+                startOffsets,
+                endOffsets,
+                directions
+            )
+        for (chunkId in chunkIds) {
+            outputSpeedLimits.get(chunkId)?.let { speedLimits -> speedLimits.add(speedLimit) }
+                ?: outputSpeedLimits.put(chunkId, mutableListOf(speedLimit))
```

Comment:
nit

```suggestion
            outputSpeedLimits.getOrPut(chunkId) { mutableListOf() }.add(speedLimit)
```

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/commit/53436ce3f4eed6b722c4714f0c69e0f989919a23